### PR TITLE
Resolve issues 228, JS error at arrow key down

### PR DESCRIPTION
**Pull request by GeorgeSokolovsky**
https://github.com/ng2-ui/auto-complete/issues/228

### CODE REVIEW

File: src/auto-complete.component.ts [hunk 1]
```diff
@@ -252,21 +252,16 @@ export class NguiAutoCompleteComponent implements OnInit {
...
-    let totalNumItem = this.filteredList.length;
-
     switch (evt.keyCode) {
       case 27: // ESC, hide auto complete
         break;
 
       case 38: // UP, select the previous li el
-        this.itemIndex = (totalNumItem + this.itemIndex - 1) % totalNumItem;
-        this.scrollToView(this.itemIndex);
+        this.handleArrowKeyEvent(true);
```
Comment (allenhwkim) [inline src/auto-complete.component.ts]: How about this ? 
```
       case 38: // UP, select the previous li el
       case 40: // DOWN, select the next li el or the first one
          this.handleArrowKeyEvent(evt)
```

File: src/auto-complete.component.ts [hunk 2]
```diff
@@ -314,4 +309,17 @@ export class NguiAutoCompleteComponent implements OnInit {
     };
   })();
 
+  private handleArrowKeyEvent(isUp: boolean) {
```
Comment (allenhwkim) [inline src/auto-complete.component.ts]: And this ? 
` private handleArrowKeyEvent(evt: Event) {`

File: src/auto-complete.component.ts [hunk 3]
```diff
@@ -314,4 +309,17 @@ export class NguiAutoCompleteComponent implements OnInit {
...
+    let totalNumItem = this.filteredList.length;
+
+    if (!totalNumItem) {
+      return;
+    }
+
+    const itemIndex = isUp ? totalNumItem + this.itemIndex - 1 : totalNumItem + this.itemIndex + 1;
```
Comment (allenhwkim) [inline src/auto-complete.component.ts]: And this ? 
` const itemIndex = evt.keyCode === 38 ? totalNumItem + this.itemIndex - 1 : totalNumItem + this.itemIndex + 1;`

File: src/auto-complete.component.ts (hunk 1 above)
Comment (GeorgeSokolovsky) [inline src/auto-complete.component.ts]: Agree with you, I'll update pr later

**Pull request closed without merging**